pkg/api/unversioned/group_version.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package unversioned
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+// TODO: We need to remove the GroupVersion in types.go. We use the name GroupAndVersion here temporarily.
+type GroupAndVersion struct {
+	Group   string
+	Version string
+}
+
+func (gv *GroupAndVersion) String() string {
+	// special case of "v1" for backward compatibility
+	if gv.Group == "" {
+		return gv.Version
```

Comment:
I really don't like this.  I'd vastly prefer to update the code to handle a properly parsing name of "/v1"